lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -382,11 +382,17 @@ sub _compose_job_overview_search_args {
 
     # determine build number
     if (!$search_args{build}) {
-        # yield the latest build of the first group if (multiple) groups but not build specified
-        # note: the search arg 'groupid' is ignored by complex_query() because we later assign 'groupids'
-        $search_args{groupid} = $groups[0]->id if (@groups);
-
-        $search_args{build} = $schema->resultset('Jobs')->latest_build(%search_args);
+        if (@groups){
+            my @builds;
+            @builds = map {
+                $search_args{groupid} = $_->{_column_data}->{id};
```

Comment:
Please don't use internal data of objects, especially not those which have an underscore in front :)

```suggestion
                $search_args{groupid} = $_->id;
```

Reply:
I think I tested this before and didn't work. I'll check it again

Reply:
I am not sure, but this doesn't work for me. the test runs but not in a browser

Reply:
What do you mean with "it doesn't work"?
It works for me, I get the expected result.
Do you get an error message?
Did you trace the SQL statements with DBIC_TRACE?

Reply:
Solved with your next suggestion